pkg/module/parse/template_to_module.go

```diff
@@ -81,7 +92,7 @@ func (p *Parser) GenerateMandatoryModulesFromTemplates(ctx context.Context,
 	return modules
 }
 
-func (p *Parser) appendModuleWithInformation(module v1beta2.AvailableModule, kyma *v1beta2.Kyma,
+func (p *Parser) appendModuleWithInformation(ctx context.Context, module v1beta2.AvailableModule, kyma *v1beta2.Kyma,
```

Comment:
Let's refactor this part a bit!
The reason to pass modules list as arg here is to append to it. Appending can be done by consumer, let's change the result of this function to return only the module. and split responsibilities. Then this func also needs a new name.

Also we may try to move all the logic from `p.newManifestFromTemplate()` into a separate dependency, let's say a `ManifestCreator` that may be tested separately.